btree/btree.c

```diff
@@ -3865,168 +3907,552 @@ int64_t btree_iter_tick(struct m0_sm_op *smop)
+	}
+	case P_ACT: {
+		struct m0_btree_rec rec;
+		struct slot         node_slot;
+		/**
+		 *  if key exists, delete the key, if there is an underflow, go
+		 *  to resolve function else return P_CLEANUP.
+		*/
+
+		if (!oi->i_key_found)
+			rec.r_flags = M0_BSC_KEY_NOT_FOUND;
+		else {
+			lev = &oi->i_level[oi->i_used];
+			node_slot.s_node = lev->l_node;
+			node_slot.s_idx  = lev->l_idx;
+			node_del(node_slot.s_node, node_slot.s_idx, bop->bo_tx);
+			lev->l_node->n_skip_rec_count_check = true;
+			node_done(&node_slot, bop->bo_tx, true);
+			node_fix(node_slot.s_node, bop->bo_tx);
+
+			rec.r_flags = M0_BSC_SUCCESS;
+		}
+		int rc = bop->bo_cb.c_act(&bop->bo_cb, &rec);
+		if (rc) {
+			lock_op_unlock(tree);
+			return fail(bop, rc);
+		}
+
+		if (oi->i_key_found) {
+			if (oi->i_used == 0 ||
+			    !node_isunderflow(lev->l_node, false)) {
+				/* No Underflow */
+				lev->l_node->n_skip_rec_count_check = false;
+				lock_op_unlock(tree);
+				return P_CLEANUP;
+			}
+			lev->l_node->n_skip_rec_count_check = false;
+			return btree_del_resolve_underflow(bop);
+		}
+		lock_op_unlock(tree);
+		return P_CLEANUP;
+	}
+	case P_FREENODE : {
+		struct nd *node;
+		int64_t    nxt = P_FREENODE;
+
+		lev = &oi->i_level[oi->i_used];
+		if (lev->l_freenode) {
+			if (oi->i_used == 0) {
+				oi->i_used = bop->bo_arbor->t_height - 1;
+				nxt = P_CLEANUP;
+			} else
+				oi->i_used --;
+
+			node = lev->l_node;
+			node_put(lev->l_node);
+			lev->l_node = NULL;
+			oi->i_nop.no_opc = NOP_FREE;
+			return node_free(&oi->i_nop, node,
+					 bop->bo_tx, nxt);
+		}
+		oi->i_used = bop->bo_arbor->t_height - 1;
+		/* Fall through */
+	}
+	case P_CLEANUP : {
+		level_cleanup(oi, bop->bo_tx);
+		return P_DONE;
+		//return m0_sm_op_ret(&bop->bo_op);
+	}
+	default:
+		M0_IMPOSSIBLE("Wrong state: %i", bop->bo_op.o_sm.sm_state);
+	};
+}
+
+int  m0_btree_open(void *addr, int nob, struct m0_btree **out)
+{
+	return 0;
+}
+
+void m0_btree_close(struct m0_btree *arbor)
+{
+}
+
+/**
+ * m0_btree_create is the API which is used by motr to create btree.
+ *
+ * @param addr It is the address of root node allocated by the caller of this
+ * function
+ * @param nob It is the size of root node.
+ * @param bt It is the type of btree to be created.
+ * @param tx It represents the transaction of which the current operation is
+ * part of.
+ * @param bop It represents the structure containing all the relevant details
+ * for carrying out btree operations.
+ */
+
+void m0_btree_create(void *addr, int nob, const struct m0_btree_type *bt,
+		     const struct node_type *nt, struct m0_be_tx *tx,
+		     struct m0_btree_op *bop)
+{
+	bop->b_data.addr	= addr;
+	bop->b_data.num_bytes	= nob;
+	bop->b_data.bt		= bt;
+	bop->b_data.nt		= nt;
+
+	m0_sm_op_init(&bop->bo_op, &btree_create_tick, &bop->bo_op_exec,
+		      &btree_conf, &bop->bo_sm_group);
+}
+
+void m0_btree_destroy(struct m0_btree *arbor, struct m0_btree_op *bop)
+{
+	bop->bo_arbor	= arbor;
+	bop->bo_tx	= NULL;
```

Comment:
Use of tabs after variable bop->bo_tx and bop->bo_arbor.
Change it to spaces please.